azurerm/internal/services/servicebus/internal.go

```diff
@@ -0,0 +1,112 @@
+package servicebus
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+func expandAuthorizationRuleRights(d *schema.ResourceData) *[]servicebus.AccessRights {
+	rights := make([]servicebus.AccessRights, 0)
+
+	if d.Get("listen").(bool) {
+		rights = append(rights, servicebus.Listen)
+	}
+
+	if d.Get("send").(bool) {
+		rights = append(rights, servicebus.SendEnumValue)
+	}
+
+	if d.Get("manage").(bool) {
+		rights = append(rights, servicebus.Manage)
+	}
+
+	return &rights
+}
+
+func flattenAuthorizationRuleRights(rights *[]servicebus.AccessRights) (listen, send, manage bool) {
+	// zero (initial) value for a bool in go is false
+
+	if rights != nil {
+		for _, right := range *rights {
+			switch right {
+			case servicebus.Listen:
+				listen = true
+			case servicebus.SendEnumValue:
+				send = true
+			case servicebus.Manage:
+				manage = true
+			default:
+				log.Printf("[DEBUG] Unknown Authorization Rule Right '%s'", right)
+			}
+		}
+	}
+
+	return listen, send, manage
+}
+
+func authorizationRuleSchemaFrom(s map[string]*schema.Schema) map[string]*schema.Schema {
+	authSchema := map[string]*schema.Schema{
+		"listen": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"send": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"manage": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"primary_key": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"primary_connection_string": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"secondary_key": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"secondary_connection_string": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+	}
+	return azure.MergeSchema(s, authSchema)
+}
+
+func authorizationRuleCustomizeDiff(d *schema.ResourceDiff, _ interface{}) error {
+	listen, hasListen := d.GetOk("listen")
+	send, hasSend := d.GetOk("send")
+	manage, hasManage := d.GetOk("manage")
+
+	if !hasListen && !hasSend && !hasManage {
+		return fmt.Errorf("One of the `listen`, `send` or `manage` properties needs to be set")
+	}
+
+	if manage.(bool) && !listen.(bool) && !send.(bool) {
```

Comment:
Based on the error message, should this not be?
```suggestion
	if manage.(bool) && (!listen.(bool) || !send.(bool)) {
```